sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -665,14 +699,14 @@ pi_result piDeviceRetain(pi_device Device) {
 
 pi_result piDeviceRelease(pi_device Device) {
   assert(Device);
-
   // TODO: OpenCL says root-device ref-count remains unchanged (1),
   // but when would we free the device's data?
-  if (--(Device->RefCount) == 0) {
-    // Destroy the command list used for initializations
-    ZE_CALL(zeCommandListDestroy(Device->ZeCommandListInit));
-    delete Device;
-  }
+  --(Device->RefCount);
```

Comment:
This is going to lead to negative refcount since we use the same PI device handle from the cache. Shouldn't we "retain" the PI devices in cache in piDevicesGet?

Reply:
In fact, SYCL RT calls piDeviceRetain() for each pi_device after it receives a vector of pi_devices from piDevicesGet().
So, it is correct to decrease the refcount.
You can see my changes in device_impl.cpp in this PR.

By the way, I have added "assert(Device->RefCount > 0)" but it is removed by someone.
I will add it back to avoid the ref count issue.

Reply:
OK, this is good.